ako-crd-operator/api/v1alpha1/healthmonitor_types.go

```diff
@@ -42,6 +42,8 @@ const (
 
 // HealthMonitorSpec defines the desired state of HealthMonitor
 // +kubebuilder:validation:XValidation:rule="(!has(self.http_monitor) || !has(self.http_monitor.auth_type) || has(self.authentication.secret_ref))",message="If auth_type is set, secret_ref must be set in authentication"
+// +kubebuilder:validation:XValidation:rule="has(oldSelf.monitor_port) == has(self.monitor_port)",message="monitor_port is immutable"
```

Comment:
is this comparing the value of monitor_port ?

Reply:
warning message getting displayed is somehow not in sync with the validation since validation checks if it is present or not and message says it is immutable